### PR TITLE
Expand the contribution guide

### PR DESCRIPTION
**Pull request by zabil**
* Break down the contribution guide to section
* Link to sources
* Add more ways to contribute

Signed-off-by: Zabil Cheriya Maliackal <zabilcm@gmail.com>

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,46 +1,118 @@
...
+extend Taiko's functionality. It is also a way to experiment with ideas which could be a part of the
+core. 
+
+You can refer the current list of [plugins](https://docs.taiko.dev/plugins/) to build your own. If you
+want your plugin to be listed on the [plugins](https://docs.taiko.dev/plugins/) page please modify the
+[list](https://github.com/getgauge/taiko/blob/master/docs/plugins.md) and send a pull request. 
+
+## Contribute code and pull requests to Taiko! 
+
+Taiko is a [Node.js](https://nodejs.org/en/) library written in Javascript. You only need
```
Comment (sriv) [inline docs/CONTRIBUTING.md:81]:
```suggestion
Taiko is a [Node.js](https://nodejs.org/en/) library written in Javascript (with Typescript bindings). You only need 
```

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,46 +1,118 @@
...
+core. 
+
+You can refer the current list of [plugins](https://docs.taiko.dev/plugins/) to build your own. If you
+want your plugin to be listed on the [plugins](https://docs.taiko.dev/plugins/) page please modify the
+[list](https://github.com/getgauge/taiko/blob/master/docs/plugins.md) and send a pull request. 
+
+## Contribute code and pull requests to Taiko! 
+
+Taiko is a [Node.js](https://nodejs.org/en/) library written in Javascript. You only need 
+the latest version of Node.js installed to clone the [repository](https://github.com/getgauge/taiko)
```
Comment (sriv) [inline docs/CONTRIBUTING.md:82]:
```suggestion
the latest version of Node.js and git installed to clone the [repository](https://github.com/getgauge/taiko)
```

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -1,46 +1,118 @@
...
+
+Taiko's [plugin](https://github.com/getgauge/taiko/wiki/Taiko-Plugin) architecture, allows you to 
+extend Taiko's functionality. It is also a way to experiment with ideas which could be a part of the
+core. 
+
+You can refer the current list of [plugins](https://docs.taiko.dev/plugins/) to build your own. If you
+want your plugin to be listed on the [plugins](https://docs.taiko.dev/plugins/) page please modify the
+[list](https://github.com/getgauge/taiko/blob/master/docs/plugins.md) and send a pull request. 
+
+## Contribute code and pull requests to Taiko!
```
Comment (sriv) [inline docs/CONTRIBUTING.md:79]: Worth referring to some tech docs? We should add those if they don't exist

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -1,46 +1,118 @@
...
 
-If your contribution is a code contribution and you do send us a pull request, please 
-sign all your commits using git.
+You can use the `-s` flag to your `git commit` command. To add this line. If you want to automate even this flag addition, you can use [git commit templates](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---templateltfilegt) which 
+automatically certifies all your future commits by default.
 
-## Bumping version of taiko
+### Bumping version of taiko
 
+Any pull request tagged with the label
```
Comment (sriv) [inline docs/CONTRIBUTING.md:115]: this looks incomplete

**Review by sriv — CHANGES_REQUESTED** [review]

File: docs/CONTRIBUTING.md (hunk 3 above)
Comment (zabil) [reply docs/CONTRIBUTING.md:79]: It's there in the para before this one.
```
Taiko's [plugin](https://github.com/getgauge/taiko/wiki/Taiko-Plugin) architecture, allows you to 
 extend Taiko's functionality. It is also a way to experiment with ideas which could be a part of the
 core. Or did you mean something else?
```

Comment (sriv) [reply docs/CONTRIBUTING.md:79]: no that's good enough i think.

**Comment by zabil** [comment]
FYI the DCO, signing is a bit messed up because I committed suggestions. And because I updated the branch it will mess the commit history if I change. If you squash and merge it should be ok.

**Review by NivedhaSenthil — APPROVED** [review]

**Pull request merged**